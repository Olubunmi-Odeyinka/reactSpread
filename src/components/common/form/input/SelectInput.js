import React from 'react';
import PropTypes from 'prop-types';
import {getFieldNameFromName } from '../../../../utilities/fieldName';

const SelectInput = ({name, options, focus, onChange, rowValue}) => {
  return (
    <div className="form-group">
      <div className="field">
        <select
          name={name}
          className="form-control"
          value={rowValue[getFieldNameFromName(name)]}
          onChange={onChange}
          autoFocus={focus}>

          <option value=""></option>
          {options && options.map((option) => {
            return <option key={option.value} value={option.value}>{option.text}</option>;
          })
          }
        </select>
      </div>
    </div>
  );
};

SelectInput.propTypes = {
  name: PropTypes.string.isRequired,
  label: PropTypes.string.isRequired,
  touched: PropTypes.bool,
  error: PropTypes.string,
  options: PropTypes.array,
  focus: PropTypes.bool,
  readOnly: PropTypes.bool
};

export default SelectInput;
