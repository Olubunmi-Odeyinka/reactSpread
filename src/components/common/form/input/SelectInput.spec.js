import React from 'react';
import { shallow } from 'enzyme';
import SelectInput from './SelectInput';

let props, wrapper;
beforeEach(() => {
  props = {
    name: 'Name',
    label: 'Name',
    touched: false,
    onChange: jest.fn(),
    rowValue: '',
    options: []
  };
  wrapper = shallow(<SelectInput {...props} />);
});

test('The dropdown component input component must render properly', ()=>{
  expect(wrapper).toMatchSnapshot();
});

// test("The dropdown component label should be 'Name' as set in props", ()=>{
//   expect(wrapper.find('label').text()).toEqual('Name');
// });

// test("The dropdown component name should be 'Name' as set in props", ()=>{
//   expect(wrapper.find('Field').props().name).toBe('Name');
// });
//
// test("The dropdown component readOnly should be 'false' as set in props", ()=>{
//   expect(wrapper.find('Field').props().readOnly).toBe(false);
// });
//
// test("The dropdown component autoFocus should be 'true' as set in props", ()=>{
//   expect(wrapper.find('Field').props().readOnly).toBe(false);
// });
//
// test("The dropdown component should have always have one extra default option added to the supplied options", ()=>{
//   props.options = [{value: 2, text: 'Satire'},
//                     {value: 3, text: 'Chain Taste'}];
//   wrapper = shallow(<SelectInput {...props} />);
//   expect(wrapper.find('option').length).toBe(3);
// });
