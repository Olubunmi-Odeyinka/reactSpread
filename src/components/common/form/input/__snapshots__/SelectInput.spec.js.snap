// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The dropdown component input component must render properly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SelectInput
    label="Name"
    name="Name"
    onChange={[Function]}
    options={Array []}
    rowValue=""
    touched={false}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="field"
>
        <select
                autoFocus={undefined}
                className="form-control"
                name="Name"
                onChange={[Function]}
                value={undefined}
        >
                <option
                        value=""
                />
        </select>
</div>,
      "className": "form-group",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <select
          autoFocus={undefined}
          className="form-control"
          name="Name"
          onChange={[Function]}
          value={undefined}
>
          <option
                    value=""
          />
</select>,
        "className": "field",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "autoFocus": undefined,
          "children": Array [
            <option
              value=""
/>,
            Array [],
          ],
          "className": "form-control",
          "name": "Name",
          "onChange": [Function],
          "value": undefined,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "option",
          },
        ],
        "type": "select",
      },
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="field"
>
          <select
                    autoFocus={undefined}
                    className="form-control"
                    name="Name"
                    onChange={[Function]}
                    value={undefined}
          >
                    <option
                              value=""
                    />
          </select>
</div>,
        "className": "form-group",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <select
            autoFocus={undefined}
            className="form-control"
            name="Name"
            onChange={[Function]}
            value={undefined}
>
            <option
                        value=""
            />
</select>,
          "className": "field",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "autoFocus": undefined,
            "children": Array [
              <option
                value=""
/>,
              Array [],
            ],
            "className": "form-control",
            "name": "Name",
            "onChange": [Function],
            "value": undefined,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "option",
            },
          ],
          "type": "select",
        },
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
